app/src/main/java/com/univoice/core_ui/CustomSpinner.kt

```diff
@@ -0,0 +1,78 @@
+package com.univoice.core_ui.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.Spinner
+import androidx.appcompat.widget.AppCompatSpinner
+
+class CustomSpinner @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : AppCompatSpinner(context, attrs, defStyleAttr) {
+    /**
+     * An interface which a client of this Spinner could use to receive
+     * open/closed events for this Spinner.
+     */
+    interface OnSpinnerEventsListener {
+        /**
+         * Callback triggered when the spinner was opened.
+         */
+        fun onSpinnerOpened(spinner: Spinner?)
+
+        /**
+         * Callback triggered when the spinner was closed.
+         */
+        fun onSpinnerClosed(spinner: Spinner?)
+    }
+
+    private var mListener: OnSpinnerEventsListener? = null
+    private var mOpenInitiated = false
+
+
+    // implement the Spinner constructors that you need
+    override fun performClick(): Boolean {
+        // register that the Spinner was opened so we have a status
+        // indicator for when the container holding this Spinner may lose focus
+        mOpenInitiated = true
+        if (mListener != null) {
+            mListener!!.onSpinnerOpened(this)
+        }
```

Comment:
P2: mListener가 null이 아니라는 조건이 위에 있어서 !! 연산자가 NullPointerException을 발생시킬 가능성이 거의 없을 것 같아요!

요런식으로 바꾸면 더 간결하고 직관적일 것 같습니다!

```suggestion
mListener?.onSpinnerOpened(this)
```